### PR TITLE
Add custom protocol for dev

### LINKED ISSUE #1879: Register a custom protocol handler just for authentication
Right now our oauth apps (dev and prod) use `x-github-client://` when terminating the web OAuth flow. This leads to a race where the last app launched (on Windows, undefined on Mac) wins making it tricky to run the dev build and prod build side-by-side while testing authentication.

We should register specialized protocol handlers just for auth that differ depending on whether you use the dev build or the prod build.

Something like `x-github-desktop-auth` and `x-github-desktop-dev-auth`. We should start registering the protocols as soon as possible and then make the switch in our oauth apps. Note that this will affect GHE as well so we'll have to keep supporting oauth callback over the generic protocol for quite some time.

### PR DESCRIPTION
**Pull request by iAmWillShepherd**
Fixes #1879

### CODE REVIEW

**Comment by iAmWillShepherd** [comment]
@niik  is this what you had in mind?

File: app/src/main-process/main.ts [hunk 1]
```diff
@@ -156,7 +156,13 @@ app.on('ready', () => {
 
   readyTime = now() - launchTime
 
-  app.setAsDefaultProtocolClient('x-github-client')
+   if (__DEV__) {
+    app.setAsDefaultProtocolClient('x-github-desktop-dev-auth')
+  } else {
+    app.setAsDefaultProtocolClient('x-github-desktop-auth')
+    app.setAsDefaultProtocolClient('x-github-client')
```
Comment (niik) [inline app/src/main-process/main.ts]: We should always register for this protocol regardless of whether we're in dev or not

Comment (iAmWillShepherd) [inline app/src/main-process/main.ts]: 👌

**Comment by iAmWillShepherd** [comment]
💯

File: app/src/main-process/main.ts [hunk 2]
```diff
@@ -156,7 +156,14 @@ app.on('ready', () => {
 
   readyTime = now() - launchTime
 
-  app.setAsDefaultProtocolClient('x-github-client')
+   app.setAsDefaultProtocolClient('x-github-client')
```
Comment (niik) [inline app/src/main-process/main.ts]: Seems to be an extra space in the indentation here

File: app/src/main-process/main.ts [hunk 3]
```diff
@@ -156,7 +156,14 @@ app.on('ready', () => {
...
+
+   if (__DEV__) {
```
Comment (niik) [inline app/src/main-process/main.ts]: Seems like an extra space snuck in here as well

Comment (iAmWillShepherd) [inline app/src/main-process/main.ts]: ⚡️

**Pull request merged**